extensions/wikia/FlowTracking/FlowTracking.hooks.php

```diff
@@ -6,4 +6,63 @@ public static function onBeforePageDisplay( \OutputPage $out, \Skin $skin ) {
 
 		return true;
 	}
+
+	/**
+	 * @param  Page     $page          The created page object
+	 * @param  User     $user          The user who created the page
+	 * @param  string   $text          Text of the new article
+	 * @param  string   $summary       Edit summary
+	 * @param  int      $minoredit     Minor edit flag
+	 * @param  boolean  $watchThis     Whether or not the user should watch the page
+	 * @param  null     $sectionAnchor Not used, set to null
+	 * @param  int      $flags         Flags for this page
+	 * @param  Revision $revision      The newly inserted revision object
+	 * @return bool
+	 */
+	public static function onArticleInsertComplete( Page $page, User $user, $text, $summary, $minoredit,
+													$watchThis, $sectionAnchor, &$flags, Revision $revision ) {
+		$title = $revision->getTitle();
+		if ( $title && $title->inNamespace( NS_MAIN ) ) {
+			$params = [];
+			$request = RequestContext::getMain()->getRequest();
+			$headers = $request->getAllHeaders();
+
+			// transforms "a=1&b=2&c=3" into [ 'a' => 1, 'b' => 2, 'c' => 3 ]
+			if ( isset( $headers[ 'REFERER' ] ) ) {
+				$params = static::getParamsFromUrlQuery( $headers[ 'REFERER' ] );
+			} else {
+				Wikia\Logger\WikiaLogger::instance()->warning( 'Flow Tracking - Referer header is not set', [
+					'useragent' => $headers[ 'USER-AGENT' ]
+				] );
+			}
+			if ( isset( $params['flow'] ) ) {
```

Comment:
I somehow don't like the fact that this check cannot yield true if https://github.com/Wikia/app/pull/11564/files#diff-b8dd1db85320c104bd4af368cc37cea6R31 was false.